plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/listener/QueryInsightsListener.java

```diff
@@ -69,11 +80,49 @@ public QueryInsightsListener(final ClusterService clusterService, final QueryIns
                 v -> this.queryInsightsService.getTopQueriesService(MetricType.LATENCY).setWindowSize(v),
                 v -> this.queryInsightsService.getTopQueriesService(MetricType.LATENCY).validateWindowSize(v)
             );
+        clusterService.getClusterSettings()
+            .addSettingsUpdateConsumer(TOP_N_CPU_QUERIES_ENABLED, v -> this.setEnableTopQueries(MetricType.CPU, v));
```

Comment:
Probably, I missed reviewing the implementation of `setEnableTopQueries` method in previous PR. Is the implementation safe for concurrent setting updates?

Reply:
yeah I think so, the concurrent logic should be handled as part of the update consumer.

Reply:
Actually, it is because the settings update consumer runs as part of ClusterApplierThread which is single threaded. Hence, we should be okay here!